zipline/pipeline/mixins.py

```diff
@@ -148,29 +152,38 @@ def _allocate_output(self, windows, shape):
             out = full(shape, missing_value, dtype=self.dtype)
         return out
 
+    def _format_inputs(self, windows, column_mask):
+        inputs = []
+        for input_ in windows:
+            window = next(input_)
+            if window.shape[1] == 1:
+                # Do not mask single-column inputs.
+                inputs.append(window)
+            else:
+                inputs.append(window[:, column_mask])
+        return inputs
+
     def _compute(self, windows, dates, assets, mask):
         """
         Call the user's `compute` function on each window with a pre-built
         output array.
         """
         compute = self.compute
         params = self.params
-        out = self._allocate_output(windows, mask.shape)
+
+        shape = (len(mask), 1) if self.ndim == 1 else mask.shape
+        out = self._allocate_output(windows, shape)
 
         with self.ctx:
             for idx, date in enumerate(dates):
-                col_mask = mask[idx]
-                masked_out = out[idx][col_mask]
+                # Never apply a mask to 1D outputs.
+                col_mask = array([True]) if self.ndim == 1 else mask[idx]
                 masked_assets = assets[col_mask]
+                out_row = out[idx][col_mask]
+                inputs = self._format_inputs(windows, col_mask)
```

Comment:
I'm a little nervous about how much work we're doing in this loop now.  On my machine a 1-year USEP spends about a second of actual execution time is just in `_format_inputs`.